db/range_tombstone_fragmenter.cc

```diff
@@ -372,4 +415,27 @@ SequenceNumber FragmentedRangeTombstoneIterator::MaxCoveringTombstoneSeqnum(
   return ValidPos() && ucmp_->Compare(start_key(), user_key) <= 0 ? seq() : 0;
 }
 
+std::vector<std::unique_ptr<FragmentedRangeTombstoneIterator>>
+FragmentedRangeTombstoneIterator::SplitBySnapshot(
+    const std::vector<SequenceNumber>& snapshots) {
+  std::vector<std::unique_ptr<FragmentedRangeTombstoneIterator>> splits;
```

Comment:
It's unclear how long this returned vector will live, which could be bad if there's a big compaction with a lot of snapshots. I might change this to return a map instead (which uses the upper bound of the range as the key).